site/content/docs/main/restore-reference.md

```diff
@@ -36,57 +22,113 @@ Available Commands:
+1. The Velero client makes a call to the Kubernetes API server to create a [`Restore`](api-types/restore.md) object.
 
-#### Always Preserve NodePorts
+1. The `RestoreController` notices the new Restore object and preforms validation.
 
-It is not always possible to set nodePorts explicitly on some big clusters because of operation complexity. Official Kubernetes documents states that preventing port collisions is responsibility of the user when explicitly specifying nodePorts:
+1. The `RestoreController` fetches basic information about the backup being restored, like the [BackupStorageLocation](locations.md) (BSL). It also a tarball of the cluster resources in the backup, any volumes that will be restored using Restic, and any volume snapshots to be restored.
 
-```
-If you want a specific port number, you can specify a value in the `nodePort` field. The control plane will either allocate you that port or report that the API transaction failed. This means that you need to take care of possible port collisions yourself. You also have to use a valid port number, one that's inside the range configured for NodePort use.
+1. The `RestoreController` then extracts the tarball of backup cluster resources to the /tmp folder and preforms some pre-processing on the resources, including
 
-https://kubernetes.io/docs/concepts/services-networking/service/#nodeport
-```
+    * Sorting the resources to help Velero decide the [restore order](#resource-restore-order) to use.
 
-The clusters which are not explicitly specifying nodePorts still may need to restore original NodePorts in case of disaster. Auto assigned nodePorts most probably defined on Load Balancers which located front side of cluster.  Changing all these nodePorts on Load Balancers is another operation complexity after disaster if nodePorts are changed.
+    * Attempting to discover the resources by their Kubernetes [Group Version Resource (GVR)](https://kubernetes.io/docs/reference/using-api/api-concepts/). If a resource is not discoverable, Velero will exclude it from the restore. See more about how [Velero backs up API versions](#backed-up-api-versions)
 
-Velero has a flag to let user deciding the preservation of nodePorts. **`velero restore create`** sub command has  **`--preserve-nodeports`** flag to **preserve** Service nodePorts **always** regardless of nodePorts **explicitly specified** or **not**. This flag used for preserving the original nodePorts from backup and can be used as **`--preserve-nodeports`** or **`--preserve-nodeports=true`**
+    * Applying any configured [resource filters](resource-filtering.md).
 
-If this flag given and/or set to true, Velero does not remove the nodePorts when restoring Service and tries to use the nodePorts which written on backup.
+    * Remapping target namespace according to specified [`--namespace-mappings`](#restoring-into-a-different-namespace) restore option.
```

Comment:
Does remapping happen here, or in the next step (line 55). I seem to have it listed twice, but i'm not sure which place is correct.